### PR TITLE
Dance with Draco in the refugee center

### PR DESCRIPTION
**Pull request by Light-Wave**
#### Summary
Features "Dance with Draco in the refugee center"

#### Purpose of change
Makes the player's current dodge be readable by json.
Adds an interaction for Draco where he will judge your dancing skills based on your current dodge.

#### Describe the solution
Arithmetics can access the dodge value with `"u_val": "dodge"`.
Added documentation for this.
Draco in the refugee center can be asked to play something you can dance to. He will have 6 different reactions to your dancing depending on how high your dodge is.

#### Describe alternatives you've considered
Have the dancing take time and/or stamina, and give a mood boost

#### Testing
I've tested the dialogue with different dodge values, it seems to work as intended.
I also tested the dodge() function. At one point the player stats displayed I had dodge 12 but the function claimed I had 11, but this could be either a rounding error or an issue with when the status display gets updated. In general, this seems to work quite well.

#### Additional context
I mostly just want to be able to read dodge with json for other purposes later on.

### CODE REVIEW

**Comment by andrei8l** [comment]
Can you put this in `u_val()` instead? I was planning to keep all the single-argument vals there and revert the `u_pain()` move too.

**Comment by Light-Wave** [comment]
> Can you put this in `u_val()` instead? I was planning to keep all the single-argument vals there and revert the `u_pain()` move too.

Sure thing. So that means moving it to condition.cpp, correct?

**Comment by Light-Wave** [comment]
Updated to use `u_val()` (or `"u_val": "dodge"`) instead.

File: data/json/npcs/refugee_center/surface_refugees/NPC_Draco_Dune.json [hunk 1]
```diff
@@ -435,6 +440,90 @@
...
+        "switch": true
+      },
+      {
+        "text": "[Do a little dance]",
+        "topic": "TALK_REFUGEE_Draco_Dance_2",
+        "condition": { "math": [ "u_val('dodge')", "<", "6" ] },
+        "switch": true
+      },
+      {
+        "text": "[Perform a descent dance]",
```
Comment (TheShadowFerret) [inline data/json/npcs/refugee_center/surface_refugees/NPC_Draco_Dune.json:462]:
```suggestion
        "text": "[Perform a decent dance]",
```
image

Comment (Light-Wave) [reply data/json/npcs/refugee_center/surface_refugees/NPC_Draco_Dune.json]: Thanks!

**Pull request merged**